superset/assets/javascripts/explore/components/AdhocMetricEditPopover.jsx

```diff
@@ -101,38 +176,68 @@ export default class AdhocMetricEditPopover extends React.Component {
 
     const popoverTitle = (
       <AdhocMetricEditPopoverTitle
-        adhocMetric={this.state.adhocMetric}
+        adhocMetric={adhocMetric}
         onChange={this.onLabelChange}
       />
     );
 
-    const stateIsValid = this.state.adhocMetric.column && this.state.adhocMetric.aggregate;
-    const hasUnsavedChanges = this.state.adhocMetric.equals(this.props.adhocMetric);
+    const stateIsValid = adhocMetric.isValid();
+    const hasUnsavedChanges = adhocMetric.equals(propsAdhocMetric);
```

Comment:
is this backwards? or I'm reading it wrong? I guess I'd expect equals to return true if they're the same which would mean no unsaved changes?

Reply:
ah you're totally right- everything works because I'm also using it wrong in the render body 🤕